Source/Core/Core/HW/WiimoteEmu/Dynamics.cpp

```diff
@@ -60,81 +53,30 @@ double CalculateStopDistance(double velocity, double max_accel)
 
 namespace WiimoteEmu
 {
-Common::Vec3 EmulateShake(ControllerEmu::Buttons* const buttons_group, const double intensity,
-                          u8* const shake_step)
+void EmulateShake(PositionalState* state, ControllerEmu::Shake* const shake_group,
+                  float time_elapsed)
 {
-  // shake is a bitfield of X,Y,Z shake button states
-  static const unsigned int btns[] = {0x01, 0x02, 0x04};
-  unsigned int shake = 0;
-  buttons_group->GetState(&shake, btns);
-
-  Common::Vec3 accel;
-
-  for (std::size_t i = 0; i != accel.data.size(); ++i)
+  auto target_position = shake_group->GetState() * shake_group->GetIntensity() / 2;
+  for (std::size_t i = 0; i != target_position.data.size(); ++i)
   {
-    if (shake & (1 << i))
-    {
-      accel.data[i] = std::sin(MathUtil::TAU * shake_step[i] / SHAKE_STEP_MAX) * intensity *
-                      GRAVITY_ACCELERATION;
-      shake_step[i] = (shake_step[i] + 1) % SHAKE_STEP_MAX;
-    }
-    else
+    if (state->velocity.data[i] * std::copysign(1.f, target_position.data[i]) < 0 ||
+        state->position.data[i] / target_position.data[i] > 0.5)
     {
-      shake_step[i] = 0;
+      target_position.data[i] *= -1;
     }
   }
 
-  return accel;
-}
-
-Common::Vec3 EmulateDynamicShake(DynamicData& dynamic_data,
-                                 ControllerEmu::Buttons* const buttons_group,
-                                 const DynamicConfiguration& config, u8* const shake_step)
-{
-  // shake is a bitfield of X,Y,Z shake button states
-  static const unsigned int btns[] = {0x01, 0x02, 0x04};
-  unsigned int shake = 0;
-  buttons_group->GetState(&shake, btns);
-
-  Common::Vec3 accel;
+  const auto peak_accel_time = 1 / shake_group->GetFrequency() / 2 / 2;
```

Comment:
This `/ 2 / 2` looks odd, any particular reason why you wrote it like that?

Reply:
I guess it's just the way my head was struggling at the math. I'll change it to `/ 4`.